File: Earthshake/EarthshakeDetailsViewController.m
//
//  EarthshakeDetailsViewController.m
//  Earthshake
//
//  Created by Andrey Paunov on 2016-04-29.
//  Copyright © 2016 AP. All rights reserved.
//

#import "EarthshakeDetailsViewController.h"
#import "EarthshakeItem.h"

@interface EarthshakeDetailsViewController ()

@property (weak, nonatomic) IBOutlet UIWebView *earthshakeWebView;
@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *spinner;

@end

@implementation EarthshakeDetailsViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.earthshakeWebView.delegate = self;

    [self loadDetailsData];
}

#pragma mark - WebView delegate methods

- (void)webViewDidFinishLoad:(UIWebView *)webView
{
    [self.spinner stopAnimating];
    self.earthshakeWebView.hidden = NO;
}

- (void)webView:(UIWebView *)webView didFailLoadWithError:(NSError *)error
{
    [self.spinner stopAnimating];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
}

#pragma mark - Helper methods

- (void)loadDetailsData
{
    self.earthshakeWebView.hidden = YES;

    [self.spinner startAnimating];

    NSURL *url = [NSURL URLWithString:self.detailURLString];
    NSURLRequest *requestObject = [NSURLRequest requestWithURL:url];

    [self.earthshakeWebView loadRequest:requestObject];
}

- (void)didSelectRefresh
{
    [super didSelectRefresh];

    [self loadDetailsData];
}

@end

File: Earthshake/EarthshakeDetailsViewController.swift
//
//  EarthshakeDetailsViewController.swift
//  Earthshake
//
//  Created by Andrey on 2017-01-03.
//  Copyright © 2017 AP. All rights reserved.
//

import Foundation

class EarthshakeDetailsViewController: EarthshakeBaseViewController, UIWebViewDelegate
{
    @IBOutlet weak var earthshakeWebView: UIWebView!
    @IBOutlet weak var spinner: UIActivityIndicatorView!

    var detailURLString = ""
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
        
        earthshakeWebView.delegate = self
        
        loadDetailsData()
    }
    
    // WebView delegate methods
    
    func webViewDidFinishLoad(_ webView: UIWebView)
    {
        spinner.stopAnimating()
        
        earthshakeWebView.isHidden = false
    }

    func webView(_ webView: UIWebView, didFailLoadWithError error: Error)
    {
        spinner.stopAnimating()
    }
    
    // Helper methods
    
    func loadDetailsData()
    {
        earthshakeWebView.isHidden = true

        spinner.startAnimating()

        let url = URL(string: detailURLString)
        let requestObject: URLRequest = URLRequest(url: url!)

        earthshakeWebView.loadRequest(requestObject)
    }
    
    override func didSelectRefresh()
    {
        super.didSelectRefresh()
        
        loadDetailsData()
    }
}
